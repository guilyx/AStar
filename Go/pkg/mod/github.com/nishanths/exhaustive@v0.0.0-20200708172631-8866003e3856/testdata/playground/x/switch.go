package x

type Direction int

var (
	N                Direction = 1
	E                Direction = 2
	S                Direction = 3
	W                Direction = 4
	directionInvalid Direction = 5
)

func _a() {
	// Basic same package.

	var d Direction
	switch d {
	case N:
	case S:
	case W:
	}
}

/*
func _b() {
	// Basic external package.

	var p bar.Phylum
	switch p {
	case bar.Chordata:
	case bar.Echinodermata:
	}
}

func _j() {
	// Named import.

	var p barpkg.Phylum
	switch p {
	case barpkg.Chordata:
	case barpkg.Echinodermata:
	}
}

func _h() {
	var p bar.Phylum
	switch p {
	}
}


func _f() {
	// Multiple values in single case.

	var d Direction
	switch d {
	case E, directionInvalid, S:
	default:
	case N:
	}
}

func _g() {
	// Non top-level switch.

	var d Direction
	if true {
		switch d {
		case (N):
		case (E):
		case (W):
		}
	}
}
*/
