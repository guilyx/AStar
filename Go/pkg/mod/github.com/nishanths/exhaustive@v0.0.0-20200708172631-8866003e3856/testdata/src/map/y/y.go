// want package:"Grain:Wheat,Rice,corn"

package bar

type Grain int

const (
	Wheat = Grain(iota)
	Rice
	corn
)
