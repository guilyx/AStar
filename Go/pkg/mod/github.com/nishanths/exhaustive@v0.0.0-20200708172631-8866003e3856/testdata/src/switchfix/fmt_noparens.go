package switchfix

import "switch/y"

func _fmt_noparens() {
	var p bar.Phylum
	switch p { // want "missing cases in switch of type bar.Phylum: Mollusca"
	case bar.Chordata:
	case bar.Echinodermata:
	}
}
