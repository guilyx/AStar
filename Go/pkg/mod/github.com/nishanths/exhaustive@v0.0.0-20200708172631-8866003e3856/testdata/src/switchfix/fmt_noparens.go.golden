package switchfix

import (
	"fmt"
	"switch/y"
)

func _fmt_noparens() {
	var p bar.Phylum
	switch p { // want "missing cases in switch of type bar.Phylum: Mollusca"
	case bar.Chordata:
	case bar.Echinodermata:
	case bar.Mollusca:
		panic(fmt.Sprintf("unhandled value: %v", p))
	}
}
